namespace/http_resolver.go

```diff
@@ -0,0 +1,423 @@
+	}
+	return res
+}
+
+func parseHTMLMetaTagName(name string) (htmlMetaTagEnum, error) {
+	switch strings.TrimSpace(name) {
+	case htmlMetaTagNameScope:
+		return htmlMetaTagScope, nil
+	case htmlMetaTagNameRegistryPush:
+		return htmlMetaTagRegistryPush, nil
+	case htmlMetaTagNameRegistryPull:
+		return htmlMetaTagRegistryPull, nil
+	case htmlMetaTagNameRegistry:
+		return htmlMetaTagRegistry, nil
+	case htmlMetaTagNameIndex:
+		return htmlMetaTagIndex, nil
+	case htmlMetaTagNameNamespace:
+		return htmlMetaTagNamespace, nil
+	default:
+		return htmlMetaTagInvalid, fmt.Errorf("unsupported meta tag name=%q", name)
+	}
+}
+
+func parseHTMLMetaTag(z *html.Tokenizer, name string) (scope, []Entry, error) {
+	var (
+		args    []string
+		entries []Entry
+		tag     htmlMetaTagEnum
+		err     error
+	)
+	for {
+		attr, val, more := z.TagAttr()
+		switch string(attr) {
+		case "name":
+			if tag != htmlMetaTagInvalid {
+				return "", nil, fmt.Errorf("expected just one name attribute")
+			}
+			tag, err = parseHTMLMetaTagName(string(val))
+			if err != nil {
+				return "", nil, err
+			}
+			if actions := tag.ToActions(); len(actions) > 0 {
+				entries = make([]Entry, len(actions))
+				for i, action := range actions {
+					entries[i] = Entry{action: action}
+				}
+			}
+		case "content":
+			args = reWhitespace.Split(strings.TrimSpace(string(val)), -1)
+			if len(args) == 1 && args[0] == "" {
+				args = []string{}
+			}
+		default:
+			return "", nil, fmt.Errorf("unrecognized meta tag attribute %s", string(attr))
+		}
+		if !more {
+			break
+		}
+	}
+	if tag == htmlMetaTagInvalid {
+		return "", nil, fmt.Errorf("meta tag without name attribute")
+	}
+	if tag == htmlMetaTagScope {
+		if len(args) != 1 {
+			return "", nil, fmt.Errorf("unexpected arguments for scope meta tag: %q", strings.Join(args, " "))
+		}
+		scp, err := parseScope(args[0])
+		if err != nil {
+			return "", nil, err
+		}
+		return scp, nil, nil
+	}
+	if args == nil && tag != htmlMetaTagNamespace {
+		return "", nil, fmt.Errorf("meta tag %s is missing content", tag.String())
+	}
+	for i := range entries {
+		entries[i].args = args
+	}
+	return "", entries, nil
+}
+
+func parseHTMLHead(body io.Reader, name string) (*Entries, error) {
+	var (
+		parsedScope     scope
+		readingMetaTags = false
+		entries         = NewEntries()
+		z               = html.NewTokenizer(body)
+		err             error
+	)
+ParsingLoop:
+	for {
+		switch tokenType := z.Next(); tokenType {
+		case html.ErrorToken:
+			if z.Err() == io.EOF {
+				break ParsingLoop
+			}
+			return nil, z.Err()
+		case html.StartTagToken, html.SelfClosingTagToken:
+			switch tagName, hasAttr := z.TagName(); string(tagName) {
+			case "head":
+				if readingMetaTags {
+					return nil, fmt.Errorf("unexpected head tag")
+				}
+			case "meta":
+				if !hasAttr {
+					return nil, fmt.Errorf("meta tag empty")
+				}
+				scp, newEntries, err := parseHTMLMetaTag(z, name)
+				if err != nil {
+					return nil, err
```

Comment:
There will need to be support for the meta tags peacefully coexisting with unrelated meta tags